index.d.ts

```diff
@@ -2207,6 +2333,94 @@ declare namespace Eris {
     toJSON(props?: string[]): JSONCache;
   }
 
+  //Interaction
+  export class Interaction extends Base {
+    applicationID: string;
+    id: string;
+    token: string;
+    type: InteractionType | number;
+    version: number;
+  }
+
+  export class PingInteraction extends Interaction {
+    type: 1;
+    acknowledge(): Promise<void>;
+    pong(): Promise<void>;
+  }
+
+  export class CommandInteraction extends Interaction {
+    channelID: string;
+    data: {
+      id: string;
+      name: string;
+      resolved?: {
+        users?: {[id: string]: User};
+        members?: { [id: string]: Omit<Member, "user" | "deaf" | "mute"> };
+        roles?: { [id: string]: Role };
+        channels?: { [id: string]: PartialChannel };
+      };
+      options?: InteractionDataOptions[];
+    };
+    guildID?: string;
+    member?: Member;
+    type: 2;
+    user?: User;
+    acknowledge(flags?: number): Promise<void>;
+    createFollowup(content: string | InteractionWebhookContent): Promise<Message>;
+    createMessage(content: string | InteractionContent): Promise<void>;
+    defer(flags?: number): Promise<void>;
+    deleteMessage(messageID: string): Promise<void>;
+    deleteOriginalMessage(): Promise<void>;
+    editMessage(messageID: string, content: string | MessageWebhookContent): Promise<Message>;
+    editOriginalMessage(content: string | MessageWebhookContent): Promise<Message>;
+    getOriginalMessage(): Promise<Message>
+  }
+
+  export class ComponentInteraction extends Interaction {
+    channelID: string;
+    data: {
+      componentType: 2 | 3;
+      custom_id: string;
```

Comment:
this should be camelCased to `customID`, like the other properties around it

Reply:
Right now it isn't because #1189 also does not, i will if bsian does aswell as if not it would be confusing to users

Reply:
The reason why my components PR doesn't camelcase is because it would be an absolute nightmare trying to map every single component type to the correct value. I haven't checked what's going on here but it would be preferable if it was kept consistent with my PR.

Reply:
I figured, more work and it wouldnt work for any new components till updated